quarkchain/cluster/shard_state.py

```diff
@@ -85,6 +86,8 @@ def __init__(self, env, full_shard_id: int, db=None, diff_calc=None):
 
         # new blocks that passed POW validation and should be made available to whole network
         self.new_block_pool = dict()
+        # height -> [coinbase address] during previous blocks, ascending order
+        self.coinbase_addr_cache = dict()  # type: Dict[int, Deque[bytes]]
```

Comment:
this cache will break when there is a fork - multiple blocks with the same height

Reply:
#319 has the same problem. we should use block header hash than height.

Reply:
good catch. updated.